todo-app/src/ListItem.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import './App.css';
+
+ export default class ListItem extends React.Component {
+   constructor(props) {
+    super(props);
+    this.state = {
+      showEdit: true,
+      editToDo: {
+        description: props.description,
+        done: false
+      }
+    }
+
+  }
+
+  Edit(showEdit) {
+    if (!showEdit) {
+      return (
+        <div className='item-name'>
+        <form id="addItem"  onSubmit={this.handelsave.bind(this)}>
+          <input type="text" value={this.state.editToDo.description} onChange={this.handleTextChange.bind(this)}/>
+          <input type="submit" value="Save" />
+          <button type="button" onClick={this.handelTextClick.bind(this)} > Cancel </button>
```

Comment:
Maybe it would be more descriptive to name this `this.handleCancel`? What do you think?